### DISCUSSION TITLE
New agents not compatible with o1-mini or o1-preview ?

### DISCUSSION THREAD

**Discussion by djuillard** [danny-avila/LibreChat#4950] [Q&A]
### What happened?

I tested the last version of Agents, which works fine for GPT-4o or Claude 3.5 v2 for instance. But I get an error when trying to use it with o1-mini or o1-preview.
Here is the error I get in the logs : 
2024-12-11T17:17:22.705Z error: [api/server/controllers/agents/client.js #sendCompletion] Unhandled error type 400 Unsupported value: 'temperature' does not support 0.25 with this mo...                                                  [truncated]

Indeed, I set the temperature to 0.25, but when I tried again with temperature set to 1 (default value), I had the same error.

### Steps to Reproduce

Create an agent with o1-mini model or o1-preview model.

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
2024-12-11T17:17:22.432Z debug: [saveConvo] api/app/clients/BaseClient.js - saveMessageToDatabase #saveConvo
2024-12-11T17:17:22.705Z error: [api/server/controllers/agents/client.js #sendCompletion] Unhandled error type 400 Unsupported value: 'temperature' does not support 0.25 with this mo...                                                  [truncated]
2024-12-11T17:17:22.708Z error: [handleAbortError] AI response error; aborting request: 400 Unsupported value: 'temperature' does not support 0.25 with this model. Only the default (...                                                  [truncated]
2024-12-11T17:17:22.719Z debug: [AgentController] Request closed
```

### Screenshots

![image](https://github.com/user-attachments/assets/1873fed1-7718-4947-9eae-48b295de345e)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
o1 doesn't support a lot of parameters that are normally accepted with the chat completions endpoint.

o1 will work with agents so long as you don't have any tools selected, or any other parameters that it doesn't accept.

It's also worth mentioning, o1 will only work for OpenAI when using Agents, not Azure, as o1 used from the latter doesn't support streaming, which is a requirement here.

**Reply by djuillard** [reply]
Excellent. 
Thanks.

**Reply by th3raid0r** [reply]
@djuillard - Why did you close this. Can you open this back up until the issue is resolved?

**Reply by djuillard** [reply]
@th3raid0r - I closed it because @danny-avila gave me the reason why it did not work. When I removed the google plugin, it did work as expected. So I agree I would prefer to be able to play with the parameters, but as indicated, it is not a bug of Librechat, it is a matter of o1 chat completion endpoint.

**Reply by th3raid0r** [reply]
@djuillard - It was converted to a Q&A thread and is no longer a Github "issue".  I noticed a few bits of nuance in this conversation that we glossed over that would be super important for users to know.

Does @danny-avila also mean that the code-interpreter is impacted? If I take this thread _literally_ then I'd assume removing all tools and actions would be sufficient. 

**However this is not the case** - One needs to disable EVERYTHING, code-interpreter, file-upload in addition to the previously mentioned "Tools and Actions". 

I suppose I'll open another one, but I kinda wish I didn't have to. 

Just be aware that these Q&A threads aren't only for you, people will wind up here after encountering issues, and by closing this so early, you increase the effort required for someone else to find this crucial information.

**Reply by djuillard** [reply]
Ok, I understand your point.

**Comment by th3raid0r** [comment]
@danny-avila - It might be worthwhile to maintain a table of supported features until everything reaches parity. 

The documentation on all of this doesn't warn a user that this will happen and I suspect folks will keep bringing up the issue because of it. 

Personally, using the code-interpreter with o1 is exactly what I wanted to do. But alas, I had no way to know it isn't supported at this time.
